src/NvimClient.APIGenerator/NvimAPIGenerator.cs

```diff
@@ -127,7 +127,25 @@ private object GetExtensionType(MessagePackExtendedTypeObject msgPackExtObj)
           var genericTypeParam = uiEvent.Parameters.Any()
             ? $"<{camelCaseName}EventArgs>"
             : string.Empty;
-          return $"    public event EventHandler{genericTypeParam} {camelCaseName}Event;";
+          return $@"
+    /// <summary>
+    /// EventHandler for <c>{uiEvent.Name}</c> UI event (see corresponding
+    /// docs in `:help ui-events` in nvim).
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// var api = new NvimAPI();
+    /// api.{uiEvent} += (sender, args) =>
+    /// {{
+    ///     // `args` is of type `{camelCaseName}EventArgs`
+    ///     // Handler code goes here.
+    /// }}
+    /// // Now if this event is emitted after attaching the UI,
+    /// // the above handler code will be run.
+    /// </code>
+    /// </example>
+    /// <seealso cref=""{camelCaseName}EventHandler""/>
```

Comment:
This is just what I came up with relatively spur-of-the-moment for the docs, lmk if/how it can be improved (and assuming this is how we want to do the docs for these `EventHandler`s).